console/console-init/ui/src/modules/address-space/utils/utils.ts

```diff
@@ -21,4 +24,72 @@ const getHeaderForDeleteDialog = (selectedItems: any[]) => {
   return header;
 };
 
-export { getDetailForDeleteDialog, getHeaderForDeleteDialog };
+const isMessgaingProjectConfigurationValid = (
+  messagingProject: IMessagingProject
+) => {
+  if (messagingProject) {
+    const { name, namespace, type, plan, authService } = messagingProject;
+    if (
+      name &&
+      name.trim() !== "" &&
+      namespace &&
+      namespace.trim() !== "" &&
+      type &&
+      type.trim() !== "" &&
+      plan &&
+      plan.trim() !== "" &&
+      authService &&
+      authService.trim() !== ""
+    ) {
+      return true;
+    }
+    return false;
+  }
+};
+
+const isEnabledCertificateStep = (messagingProject: IMessagingProject) => {
+  if (messagingProject.customizeEndpoint === true) {
```

Comment:
There are so many if else conditions. It can be more simple and readable. Can you try below implementation.

const isEnabledCertificateStep = (messagingProject: IMessagingProject) => {
  if (messagingProject.customizeEndpoint === true && 
     (messagingProject.tlsCertificate !== TlsCertificateType.UPLOAD_CERT || 
       messagingProject.tlsCertificate === TlsCertificateType.UPLOAD_CERT &&
        messagingProject.privateKey &&
        messagingProject.privateKey.trim() !== "" &&
        messagingProject.certValue &&
        messagingProject.certValue?.trim() !== ""
     )
  ) {
    return true;
  } 
  return false;
};